### PR TITLE
Don't collect usage data during Fuchsia build

### PR DESCRIPTION
**Pull request by abarth**
Previously, the usage analytics would generate a write to the user's
HOME directory during the Fuchsia build. We're tightening down the
environment in which we run the Fuchsia build, and these writes are now
more obvious.

This patch removes the usage analytics during the Fuchsia build,
avoiding the write to the user's HOME directory.

### CODE REVIEW

File: packages/flutter_tools/lib/src/disabled_usage.dart [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  set suppressAnalytics(bool value) { }
+
+  @override
+  bool get enabled => false;
+
+  @override
+  set enabled(bool value) { }
+
+  @override
+  String get clientId => '00000000-0000-4000-0000-000000000000';
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/disabled_usage.dart]: Not `null`?

**Review by tvolkert — APPROVED** [review]

File: packages/flutter_tools/lib/src/disabled_usage.dart (hunk 1 above)
Comment (abarth) [reply packages/flutter_tools/lib/src/disabled_usage.dart]: I can try null.  The MockUsage class uses this value, which makes me wonder if it needs to be non-null for some reason.

Comment (abarth) [reply packages/flutter_tools/lib/src/disabled_usage.dart]: Null works :confetti_ball:

**Comment by abarth** [comment]
Build failure on the dashboard looks like a flake.

**Pull request merged**